src/Dom/ElementList.php

```diff
@@ -31,13 +32,13 @@ final class ElementList implements IteratorAggregate, Countable {
 	/**
 	 * Adds an element to the list, possibly with a caption.
 	 *
-	 * @param DOMElement $element The element to add, possibly an image.
-	 * @param string     $caption The caption to add, if any.
+	 * @param DOMElement   $element The element to add, possibly an image.
+	 * @param DOMNode|null $caption The caption for the element.
 	 * @return ElementList A clone of this list, with the new element added.
 	 */
-	public function add( DOMElement $element, $caption = '' ) {
+	public function add( DOMElement $element, DOMNode $caption = null ) {
```

Comment:
See question above regarding whether `$captions` should actually be typed as `DOMElement`.